src/ngraph_builder.cc

```diff
@@ -2233,6 +2233,66 @@ static Status TranslateDirectReduceOp(
       });
 }
 
+static Status TranslateOneHotOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  shared_ptr<ng::Node> ng_features, ng_on, ng_off;
+  TF_RETURN_IF_ERROR(
+      GetInputNodes(ng_op_map, op, &ng_features, nullptr, &ng_on, &ng_off));
+
+  auto ng_features_shape = ng_features->get_shape();
+  auto ng_features_rank = ng_features_shape.size();
+
+  std::vector<int> depth;
+  TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &depth));
+
+  // Depth must be scalar
+  if (depth.size() != 1) {
+    return errors::InvalidArgument(
+        "OneHot Op: depth of one hot dimension must be scalar ", depth.size());
+  }
+
+  int one_hot_axis;
+  TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "axis", &one_hot_axis));
+
+  std::vector<int> shape;
+  for (int i = 0; i < ng_features_rank; i++) {
+    shape.push_back(ng_features_shape[i]);
+  }
```

Comment:
ngraph's `Shape` is subclassed from `vector<size_t>`. So you can use it as a vector, no need to create a new vector out of it
https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/shape.hpp#L28

```
ng::Shape output_shape(ng_features_shape);
// declare pos
if (one_hot_axis == -1) {
one_hot_axis = ng_features_rank;
pos = output_shape.end();
} else {
pos = output_shape.begin() + one_hot_axis;
}
output_shape.insert(pos, depth[0]);
```